pipeline/trusted-resources/cmd/sign/main.go

```diff
@@ -46,10 +49,20 @@ func main() {
 
 	flag.Parse()
 
-	// Load signer from key files
-	signer, err := signature.SignerFromKeyRef(ctx, *privateKey, generate.GetPass)
-	if err != nil {
-		log.Fatalf("error getting signer: %v", err)
+	var signer sigstore.Signer
+	var err error
+	if *cosignKey != "" {
+		// Load signer from key files
+		signer, err = signature.SignerFromKeyRef(ctx, *cosignKey, generate.GetPass)
+		if err != nil {
+			log.Fatalf("error getting signer: %v", err)
+		}
+	}
+	if *kmsKey != "" {
+		signer, err = kms.Get(ctx, *kmsKey, crypto.SHA256)
+		if err != nil {
+			log.Fatalf("error getting signer: %v", err)
+		}
```

Comment:
Fine for now, but I'm curious if there's a func in the sigstore/cosign libs that will do this for us 🤔 

(if not, this might be a useful thing to add!)